modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -87,85 +63,41 @@ RadialReturnStressUpdate::updateState(RankTwoTensor & strain_increment,
-  // In that case skip the entire iteration if the effective trial stress is zero
-  if (effective_trial_stress != 0.0)
+  if (_legacy_return_mapping)
   {
-    computeStressInitialize(effective_trial_stress, elasticity_tensor);
-
-    // Use Newton iteration to determine the scalar effective inelastic strain increment
-    unsigned int iteration = 0;
-
-    Real residual, norm_residual, first_norm_residual = 0;
-
-    // create an output string with iteration information when errors occur
-    std::string iteration_output;
-
-    do
-    {
-      iterationInitialize(scalar_effective_inelastic_strain);
-
-      residual = computeResidual(effective_trial_stress, scalar_effective_inelastic_strain);
-      norm_residual = std::abs(residual);
-      if (iteration == 0)
-      {
-        first_norm_residual = norm_residual;
-        if (first_norm_residual == 0)
-          first_norm_residual = 1;
-      }
-
-      Real derivative =
-          computeDerivative(effective_trial_stress, scalar_effective_inelastic_strain);
-
-      scalar_effective_inelastic_strain -= residual / derivative;
-
-      if (_output_iteration_info || _output_iteration_info_on_error)
-      {
-        iteration_output +=
-            "In the element " + Moose::stringify(_current_elem->id()) + +" and the qp point " +
-            Moose::stringify(_qp) + ": \n" + +" iteration = " + Moose::stringify(iteration) + "\n" +
-            +" effective trial stress = " + Moose::stringify(effective_trial_stress) + "\n" +
-            +" scalar effective inelastic strain = " +
-            Moose::stringify(scalar_effective_inelastic_strain) + "\n" + +" relative residual = " +
-            Moose::stringify(norm_residual / first_norm_residual) + "\n" +
-            +" relative tolerance = " + Moose::stringify(_relative_tolerance) + "\n" +
-            +" absolute residual = " + Moose::stringify(norm_residual) + "\n" +
-            +" absolute tolerance = " + Moose::stringify(_absolute_tolerance) + "\n";
-        iteration_output += "\n";
-      }
-
-      iterationFinalize(scalar_effective_inelastic_strain);
-      ++iteration;
-    } while (iteration < _max_its && norm_residual > _absolute_tolerance &&
-             (norm_residual / first_norm_residual) > _relative_tolerance);
-
-    if (_output_iteration_info)
-      _console << iteration_output << std::endl;
-
-    if (iteration == _max_its && norm_residual > _absolute_tolerance &&
-        (norm_residual / first_norm_residual) > _relative_tolerance)
-    {
-      if (_output_iteration_info_on_error)
-        Moose::err << iteration_output;
-
-      mooseError("Exceeded maximum iterations in RadialReturnStressUpdate solve for material: ",
-                 _name,
-                 ".  Rerun with  'output_iteration_info_on_error = true' for more information.");
-    }
-
-    // compute inelastic strain increments while avoiding a potential divide by zero
+    if (effective_trial_stress < 0.01)
```

Comment:
This super bodgy code is copied straight from the solid_mechanics implementation.  I kept it there just so it would behave the same as it did while we keep the legacy option around, but since it was never there in the tensor_mechanics version, we shouldn't do it that way.  @sapitts, I thought you weren't even going to include the legacy option in tensor_mechanics. Did you end up needing it?